platform/darwin/src/NSData+MGLAdditions.mm

```diff
@@ -0,0 +1,33 @@
+#import "NSData+MGLAdditions.h"
+
+#include <mbgl/util/compression.hpp>
+
+@implementation NSData (MGLAdditions)
+
+- (NSData *)mgl_compressedData
+{
+    NSString *dataString = [[NSString alloc] initWithData:self encoding:NSUTF8StringEncoding];
+    
+    std::string *string = new std::string([dataString UTF8String]);
+    std::string compressed_string = mbgl::util::compress(*string);
+    
+    delete string;
+    
+    return [NSData dataWithBytes:(void *) compressed_string.c_str() length:compressed_string.length()];
+}
+
+- (NSData *)mgl_decompressedData
+{
+    unsigned char buffer[[self length]];
+    
+    [self getBytes:buffer length:[self length]];
```

Comment:
You can avoid this step by accessing `self.bytes`. Since `mbgl::util::decompress()` only reads that memory within the scope of this method, that otherwise unsafe property is fine to use locally.